src/VisualStudio/Core/Def/StackTraceExplorer/StackFrameViewModel.cs

```diff
@@ -0,0 +1,310 @@
+                        .WithChangedOption(new OptionKey(NavigationOptions.ActivateTab), false);
+
+                var document = _workspace.CurrentSolution.GetRequiredDocument(sourceLocation.SourceTree);
+
+                await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+                navigationService.TryNavigateToSpan(_workspace, document.Id, sourceLocation.SourceSpan, options, cancellationToken);
+            }
+            catch (Exception ex) when (FatalError.ReportAndCatchUnlessCanceled(ex, cancellationToken))
+            {
+            }
+        }
+
+        public void NavigateToSymbol()
+        {
+            var cancellationToken = _threadingContext.DisposalToken;
+            Task.Run(() => NavigateToMethodAsync(cancellationToken), cancellationToken);
+        }
+
+        public async Task NavigateToMethodAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var symbol = await _frame.ResolveSymbolAsync(_workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
+
+                if (symbol is null)
+                {
+                    // Show some dialog?
+                    return;
+                }
+
+                var success = GoToDefinitionHelpers.TryGoToDefinition(
+                    symbol,
+                    _workspace.CurrentSolution,
+                    _threadingContext,
+                    _streamingPresenter,
+                    thirdPartyNavigationAllowed: true,
+                    cancellationToken: cancellationToken);
+
+                if (!success)
+                {
+                    // Show some dialog?
+                    return;
+                }
+            }
+            catch (Exception ex) when (FatalError.ReportAndCatchUnlessCanceled(ex, cancellationToken))
+            {
+            }
+        }
+
+        public void NavigateToFile()
+        {
+            var cancellationToken = _threadingContext.DisposalToken;
+            Task.Run(() => NavigateToFileAsync(cancellationToken), cancellationToken);
+        }
+
+        public async Task NavigateToFileAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var document = GetDocument(out var lineNumber);
+
+                if (document is not null)
+                {
+                    // While navigating do not activate the tab, which will change focus from the tool window
+                    var options = _workspace.Options
+                            .WithChangedOption(new OptionKey(NavigationOptions.PreferProvisionalTab), true)
+                            .WithChangedOption(new OptionKey(NavigationOptions.ActivateTab), false);
+
+                    var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+                    lineNumber = Math.Min(sourceText.Lines.Count, lineNumber);
```

Comment:
This seems odd... navigate to the last line if one wasn't provided? Wouldn't the top of the file make more sense?

Reply:
It's not the case where one isn't provided, but if the line number is greater than current lines. Since these line numbers aren't always going to match (because versioning), it's kind of a best effort. That's why the default navigation is symbol name based.

Reply:
Ahh makes sense... Probably worth a comment